src/mongoose.ts

```diff
@@ -103,11 +128,45 @@ export class MongoosePlugin extends BasePlugin<typeof mongoose> {
     }
   }
 
+  private patchAndCaptureSpanContext(funcName: string) {
+    const thisPlugin = this
+    thisPlugin._logger.debug(`MongoosePlugin: patched mongoose query ${funcName} prototype`);
+    return (original: Function) => {
+      return function captureSpanContext(this: any) {
+        this._otContext = thisPlugin._tracer.getCurrentSpan();
+        return original.apply(this, arguments);
+      }
+    }
+  }
+
+  private patchQueryThen() {
+    const thisPlugin = this
+    thisPlugin._logger.debug('MongoosePlugin: patched mongoose query then prototype');
+    return (originalThen: Function) => {
+      return function patchedThen(this: any) {
+        if(this._otContext) {
+          return thisPlugin._tracer.withSpan(this._otContext, () => {
```

Comment:
you could replace that by `thisPlugin._tracer.bind(this._otContext, originalThen)` (interally it does almost the same but it would be clearer for you i think)

Reply:
Good comment, thank you